schemas/tls.yml

```diff
@@ -0,0 +1,174 @@
+---
+- name: tls
+  title: TLS
+  group: 2
+  short: Fields describing a TLS connection.
+  description: >
+    Fields related to TLS activity.
+  type: group
+  fields:
+
+    - name: version.protocol
+      level: extended
+      type: keyword
+      description: Normalized protocol name parsed from original string (e.g. ssl, tls).
+
+    - name: version.number
+      level: extended
+      type: keyword
+      description: Numeric part of the version parsed from the original string (e.g. 1.2, 3).
+
+    - name: cipher
+      type: keyword
+      level: extended
+      description: String indicating the cipher used during the current connection.
+
+    - name: curve
+      type: keyword
+      level: extended
+      description: String indicating the curve used for the given cipher, when applicable.
+
+    - name: resumed
+      type: boolean
+      level: extended
+      description: Boolean flag indicating if this TLS connection was resumed from an existing TLS negotiation.
+
+    - name: next_protocol
+      type: keyword
+      level: extended
+      description: String indicating the protocol being tunneled (e.g. http/1.1, spdy/3, imap, webrtc)
+
+    - name: client.ja3
+      type: keyword
+      level: extended
+      description: A hash that identifies clients based on how they perform an SSL/TLS handshake.
+
+    - name: client.server_name
+      type: keyword
+      level: extended
+      description: >
+        Also called an SNI, this tells the server which hostname to which the client is attempting to connect.
+
+    - name: client.supported_ciphers
+      type: keyword
+      level: extended
+      description: List of ciphers offered by the client during the client hello.
+
+    - name: client.subject
+      type: keyword
+      level: extended
+      description: Subject of the x.509 certificate presented by the client.
+
+    - name: client.issuer
+      type: keyword
+      level: extended
+      description: Subject of the issuer of the x.509 certificate presented by the client.
+
+    - name: client.not_before
+      type: date
+      level: extended
+      description: Timestamp indicating when client certificate is first considered valid.
+
+    - name: client.not_after
+      type: date
+      level: extended
+      description: Timestamp indicating when client certificate is no longer considered valid.
+
+    - name: client.certificate_chain
+      type: keyword
+      level: extended
+      description: List of PEM-encoded certificates that make up the certificate chain offered by the client.
```

Comment:
Like we discussed, it would be good to mention in both `*.certificate` and `*.certificate_chain` that they are more or less mutually exclusive. If the source has the cert chain fill that, and cert can be omitted. If only cert is available, then fill that one instead.

Reply:
I like that the example shows the chain as an array, good clarification.

Could you adjust the description so it's mentioned explicitly there as well? Could be as simple as replacing the word "List" with "Array".

Reply:
You know what, I'm wondering if we should really make them mutually exclusive.

If the source only gives access to the cert itself, of course populate that one.

But if the source gives access to the whole chain, I think it's worth populating both. This would ensure that the clean single value `*.certificate` is almost always populated, and therefore consumers of the data can reliably depend on its presence.

Then if the chain is also available, consumers can go dig deeper there as well. The schema can push to fill both. If users of the solution want to trim down index size, they're free to remove the chain, at the cost of losing this information.